libguard/guard_exception.hpp

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <exception>
+#include <string>
+
+class guardFileOpen : public std::exception
+{
+  public:
+    explicit guardFileOpen(const std::string& message) : message(message)
```

Comment:
Why you are expecting a string from a thrower? Why can not use that failure in reason in this class? or is it will vary from one place to another?

Reply:
For few I can add in class itself, but for some we have different strings